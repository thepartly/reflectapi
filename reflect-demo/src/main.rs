#[cfg(test)]
mod tests;

#[derive(reflect::Input)]
struct TestStructWithVec<T>
where
    T: reflect::Input,
{
    _f: T,
}

#[derive(reflect::Input)]
struct TestStructParent
// where
//     T: reflect::Input,
{
    _f: TestStructWithVec<u8>,
    // _f2: TestStructWithVec<T>,
}

// #[derive(serde::Deserialize)]
// struct Test<'a> {
//     _a: std::slice::Iter<'a, u8>,
// }

trait MyTrait {}

fn main() {
    // println!("{}", MyStruct::reflect_input());
    println!("{:#?}", TestStructParent::reflect_input());
    // println!("{:#?}", TestStructWithVec::<u8>::reflect_input());
}
