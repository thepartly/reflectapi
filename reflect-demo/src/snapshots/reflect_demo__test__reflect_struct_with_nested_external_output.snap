---
source: reflect-demo/src/main.rs
expression: "TestStructWithNestedExternal::reflect_output()"
---
Schema {
    name: "",
    types: [
        Type {
            name: "TestStructWithNestedExternal",
            fields: [
                Field {
                    name: "_f",
                    type_ref: TypeRef {
                        name: "crate::test_lib::TestStructNested",
                    },
                    _debug: "",
                },
            ],
            _debug: "reflect_demo::test",
        },
        Type {
            name: "TestStructNested",
            fields: [
                Field {
                    name: "_f",
                    type_ref: TypeRef {
                        name: "String",
                    },
                    _debug: "",
                },
            ],
            _debug: "reflect_demo::test_lib",
        },
    ],
    _debug: "",
}
